### PR TITLE
fix: Some error in storage is not handled

### PR DESCRIPTION
**Pull request by bokket**
Signed-off-by: bokket <3100563328@qq.com>

### CODE REVIEW

**Comment by bokket** [comment]
@Xuanwo Please help me review

File: storage.go [hunk 1]
```diff
@@ -34,29 +34,32 @@ func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
...
 	rp := s.getAbsPath(path)
 	dir := filepath.Dir(rp)
-	err = s.hdfs.MkdirAll(dir, 0666)
-	if err != nil {
-		return
-	}
+
+	//	If dirname is already a directory,
+	// 	MkdirAll does nothing and returns nil.
+	err = s.hdfs.MkdirAll(dir, 0755)
```
Comment (Xuanwo) [inline storage.go:40]: Please always check the errors. If we meet errors here, something must wrong and we should go further.

File: storage.go [hunk 2]
```diff
@@ -34,29 +34,32 @@ func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
...
 
 	_, err = s.hdfs.Stat(rp)
 
-	if err == nil {
-		err = s.hdfs.Remove(rp)
-		if err != nil && errors.Is(err, os.ErrNotExist) {
-			// Omit `file not exist` error here
-			// ref: [GSP-46](https://github.com/beyondstorage/specs/blob/master/rfcs/46-idempotent-delete.md)
-			err = nil
+	if err != nil && errors.Is(err, os.ErrNotExist) {
```
Comment (Xuanwo) [inline storage.go]: How about keep complex logic out of the scope?

I prefer:

```go
if err == nil {
    // remove xxx (check error please)
}
if err != nil && !errors.Is(err, os.ErrNotExist) {
   // other error happens.
}

// File is not exist, we can create now.
```

File: storage.go [hunk 3]
```diff
@@ -34,28 +34,43 @@ func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
...
 	_, err = s.hdfs.Stat(rp)
 
+	//	The error returned by Stat can only be nil or not os.ErrNotExist
 	if err == nil {
-		err = s.hdfs.Remove(rp)
-		if err != nil && errors.Is(err, os.ErrNotExist) {
-			// Omit `file not exist` error here
-			// ref: [GSP-46](https://github.com/beyondstorage/specs/blob/master/rfcs/46-idempotent-delete.md)
-			err = nil
+		// If the path does not exist,
```
Comment (Xuanwo) [inline storage.go]: If the error returned by `Stat` is nil, the path must exist. I think we don't need to explain the detailed behavior of our hdfs SDK. Instead, we need to comment on why we need to do remove it here.

File: storage.go [hunk 4]
```diff
@@ -83,12 +102,11 @@ func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCre
...
-	if err != nil && errors.Is(err, os.ErrNotExist) {
-		// Omit `file not exist` error here
-		// ref: [GSP-46](https://github.com/beyondstorage/specs/blob/master/rfcs/46-idempotent-delete.md)
-		err = nil
-	}
+
+	// 	If the path does not exist,
+	// 	RemoveAll returns nil (no error).
+	//	If there is an error here other than os.ErrNotExist is also thrown directly
+	err = s.hdfs.RemoveAll(rp)
```
Comment (Xuanwo) [inline storage.go]: `RemoveAll` is not a good idea. As we only allow remove one object at a time.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**